### PR TITLE
a couple of fixes for hardsuit helmets

### PR DESCRIPTION
**Pull request by Eniraz**
**What does this PR do:**
Hardsuit helmets now unlink when removed from the hardsuit they were attached too. It also adds a chack to prevent toggling combat mode for a syndicate hardsuit helmet while it's in hand, which presumably caused #6197, since toggling it makes the helmet NODROP and can only be fixed by putting it on a hardsuit and then putting it on.

**Images of sprite/map changes (IF APPLICABLE):**
None

**Changelog:**
:cl: Eschess
fix: hardsuit helmets now unlink from the hardsuits upon being detached
fix: syndicate hardsuit helmet combat mode can no longer be toggled on hand
fix: fixes non-syndicate hardsuit helmet attaching/detaching runtimes
/:cl:

### CODE REVIEW

File: code/modules/clothing/spacesuits/hardsuit.dm [hunk 1]
```diff
@@ -206,6 +206,9 @@
 		else
 			to_chat(user, "You detach \the [helmet] from \the [src]'s helmet mount.")
 			helmet.loc = get_turf(src)
+			if(istype(src,/obj/item/clothing/head/helmet/space/hardsuit/syndi))
```
Comment (variableundefined) [inline code/modules/clothing/spacesuits/hardsuit.dm]: No src. Why would you check the type of the src? Isn't that known already?

File: code/modules/clothing/spacesuits/hardsuit.dm [hunk 2]
```diff
@@ -206,6 +206,9 @@
 		else
 			to_chat(user, "You detach \the [helmet] from \the [src]'s helmet mount.")
 			helmet.loc = get_turf(src)
+			if(istype(src,/obj/item/clothing/head/helmet/space/hardsuit/syndi))
+				var/obj/item/clothing/head/helmet/space/hardsuit/syndi/S = helmet
+				S.linkedsuit = null
 			src.helmet = null
```
Comment (variableundefined) [inline code/modules/clothing/spacesuits/hardsuit.dm]: Missed this. Remove src., it is implicit

File: code/modules/clothing/spacesuits/hardsuit.dm [hunk 3]
```diff
@@ -227,10 +230,11 @@
 			user.drop_item()
 			W.loc = src
 			src.helmet = W
```
Comment (variableundefined) [inline code/modules/clothing/spacesuits/hardsuit.dm]: Remove src

File: code/modules/clothing/spacesuits/hardsuit.dm [hunk 4]
```diff
@@ -227,10 +230,11 @@
 			user.drop_item()
 			W.loc = src
 			src.helmet = W
-			var/obj/item/clothing/head/helmet/space/hardsuit/syndi/S = W
-			S.forceMove(src)
-			helmet = S
-			S.link_suit()
+			if(istype(src,/obj/item/clothing/head/helmet/space/hardsuit/syndi))
```
Comment (variableundefined) [inline code/modules/clothing/spacesuits/hardsuit.dm]: See above - Why typecheck src? You already know its typepath

File: code/modules/clothing/spacesuits/hardsuit.dm (hunk 2 above)
Comment (KasparoVy) [reply code/modules/clothing/spacesuits/hardsuit.dm]: mfw merged anway

**Pull request merged**